clang/lib/Sema/SemaExpr.cpp

```diff
@@ -3107,8 +3107,11 @@ Sema::PerformObjectMemberConversion(Expr *From,
                                    /*IgnoreAccess=*/true))
     return ExprError();
 
-  return ImpCastExprToType(From, DestType, CK_UncheckedDerivedToBase,
-                           VK, &BasePath);
+  if (FromType.isVolatileQualified())
+    DestType.addVolatile();
```

Comment:
Should we be doing this for all qualifiers? `const` seems like it should almost certainly behave the same way, but I think it's a more general property of qualifiers that the derived to base cast would keep the qualifiers.

Reply:
From intro.object:
> Objects can contain other objects, called subobjects. A subobject can be a member subobject a base class subobject, or an array element.

So, given:
```cpp
struct Base { int X; };
struct Derived : Base {};

const Derived Obj;
```
Seems outright logical that the entirety of Obj must be const. Thought this applied only for cv-qualifiers, but should indeed make sense for all qualifiers; updated, thanks!

Reply:
EDIT: Moving from:
```cpp
DestType = Context.getQualifiedType(DestType, FromType.getQualifiers());
```
To CVR ones for now:
```cpp
DestType.withCVRQualifiers(FromType.getCVRQualifiers());
```
Per CI failure [here](https://buildkite.com/llvm-project/github-pull-requests/builds/149797#01952e46-e199-450d-978b-2f763cd9b9c0), due to incompatibility between address spaces. Not completely sure if we should setAddressSpace of DestType AS one to the one of FromType, before getQualifiedType.